src/KahanSummation.jl

```diff
@@ -113,4 +114,6 @@ function sum_kbn(A)
     s - c
 end
 
+sum_kbn(f, A) = f.(A)
```

Comment:
This isn't quite correct; `f` is applied elementwise to `A` but the result isn't summed. Perhaps you meant `sum_kbn(f.(A))`?

Even that will make two passes over the array, so we're likely able to do better. I'm not sure this is entirely correct either, but something like this would be a start:
```diff
diff --git a/src/KahanSummation.jl b/src/KahanSummation.jl
index fde6700..26a7535 100644
--- a/src/KahanSummation.jl
+++ b/src/KahanSummation.jl
@@ -91,22 +91,23 @@ end
 Return the sum of all elements of `A`, using the Kahan-Babuska-Neumaier compensated
 summation algorithm for additional accuracy.
 """
-function sum_kbn(A)
+function sum_kbn(f, A)
     T = @default_eltype(typeof(A))
-    c = promote_sys_size_add(zero(T)::T)
+    c = f(promote_sys_size_add(zero(T)::T))
     i = start(A)
     if done(A, i)
         return c
     end
     Ai, i = next(A, i)
-    s = Ai - c
+    s = f(Ai) - c
     while !(done(A, i))
         Ai, i = next(A, i)
-        t = s + Ai
-        if abs(s) >= abs(Ai)
-            c -= ((s-t) + Ai)
+        fAi = f(Ai)
+        t = s + fAi
+        if abs(s) >= abs(fAi)
+            c -= ((s-t) + fAi)
         else
-            c -= ((Ai-t) + s)
+            c -= ((fAi-t) + s)
         end
         s = t
     end
```

Reply:
is `view` applicable?

Reply:
Hm, not exactly, I don't think, since the issue is iterating over the input multiple times, whereas `view` would be useful to avoid making copies.